x-pack/plugins/stack_connectors/server/connector_types/lib/servicenow/service.ts

```diff
@@ -249,6 +255,72 @@ export const createExternalService: ServiceFactory = ({
     }
   };
 
+  const getIncidentByCorrelationId = async (correlationId: string): Promise<ServiceNowIncident> => {
+    try {
+      const res = await request({
+        axios: axiosInstance,
+        url: getIncidentByCorrelationIdUrl(correlationId),
+        method: 'get',
+        logger,
+        configurationUtilities,
+      });
+
+      checkInstance(res);
+
+      if (res.status === 404) {
```

Comment:
By default `axios` will throw an error for status codes > 300 ([`validateStatus](https://github.com/axios/axios)`). Did you notice a different behavior? Also because this is a `find` request I think `res.data.result` will be empty and ServiceNow will not return a `404` error when the search does match our criteria.

Reply:
hmm.. 
Then I will put a check directly in closeIncident when `incidentToBeClosed` is null we log that we didn't find any incident.